sci-physics/fastjet/fastjet-3.4.0.ebuild

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+FORTRAN_NEEDED=plugins
+PYTHON_COMPAT=( python3_{8..10} )
+
+inherit autotools flag-o-matic fortran-2 python-r1
+
+DESCRIPTION="A software package for jet finding in pp and e+e- collisions"
+HOMEPAGE="http://fastjet.fr/"
+SRC_URI="http://fastjet.fr/repo/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="cgal doc examples python +plugins +contrib"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="
+sys-devel/gcc:=[fortran]
+contrib? ( sci-physics/fastjet-contrib )
+cgal? ( sci-mathematics/cgal:=[shared(+)] )
+plugins? ( sci-physics/siscone:= )
+python? ( ${PYTHON_DEPS} )
+"
+RDEPEND="${DEPEND}"
+BDEPEND="doc? ( app-doc/doxygen[dot] )"
+
+PATCHES=( 
+	"${FILESDIR}"/${P}-system-siscone.patch 
+	"${FILESDIR}"/${P}-gfortran.patch 
+	)
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	use cgal && has_version 'sci-mathematics/cgal[gmp]' && append-libs -lgmp
```

Comment:
is `has_version` necessary here? We DEPEND on it with USE=cgal so we can safely assume that it is present

Reply:
I thought so too, but I decided to follow the gentoo-tree version: https://gitweb.gentoo.org/repo/gentoo.git/tree/sci-physics/fastjet/fastjet-3.0.6-r3.ebuild

Reply:
> I thought so too, but I decided to follow the gentoo-tree version: https://gitweb.gentoo.org/repo/gentoo.git/tree/sci-physics/fastjet/fastjet-3.0.6-r3.ebuild

That an ebuild is in ::gentoo does not necessarily mean that it is a perfect example, there are also some ebuilds in there that are not quiet up to spec. That being said, I think that in this case it can't hurt to leave that check in there, at worst it does nothing useful.

Reply:
I think cgal can have gmp enabled and disabled. In one case we need to add -lgmp.

Reply:
> I think cgal can have gmp enabled and disabled. In one case we need to add `-lgmp`.

Probably, but that opens up a new can of worms. If I have cgal[gmp] installed when I emerge this ebuild, but then at some later time re-install cgal without the gmp flag, does fastjet break?
If yes then we need to have a USE flag to toggle a dependency on cgal[gmp] to ensure that the above cannot happen because portage will complain. If no then this is fine.

Reply:
Possibly related bug report: https://bugs.gentoo.org/851912

Reply:
> Possibly related bug report: https://bugs.gentoo.org/851912

The problem here seems to be that in version 5.5 of cgal the gmp USE flag has been dropped and it instead depends on `dev-libs/gmp` unconditionally. This means that with version 5.5 the `has_version` check will return False when it should return True.

We can do two things, either we need specific cases of this check for versions <5.5 of cgal and for versions >=5.5. Or we just depend on `sci-mathematics/cgal[gmp(+)]` and drop the check entirely since then we can assume that the gmp flag is always enabled on cgal.

Reply:
unstable cgal-5.5 removed the gmp flag, but still depends on the lib. So it depends on if the cgal defaults are with `-DWITH_GMP=ON`...

Reply:
CC'ing the author of cgal-5.5 @thesamesam 

I'm confused about this, it looks weird to me. And then there is https://bugs.gentoo.org/851912

Reply:
I decided to depend on `sci-mathematics/cgal[gmp(-)]`, which filters cgal-5.5 away until it is resolved.